CMakeLists.txt

```diff
@@ -14,53 +14,46 @@ include(GNUInstallDirs)
 set(CMAKE_MODULE_PATH ${CMAKE_CURRENT_SOURCE_DIR}/cmake)
 
 ##---------------------------------------------------------------------------##
-# Set up main options
+# Set up main options (inherit from Kokkos and Cabana CMake)
 ##---------------------------------------------------------------------------##
+find_package(Cabana REQUIRED)
+find_package(MPI REQUIRED)
+
 option(CabanaMD_ENABLE_Serial "Build CabanaMD with Serial support" ON)
 if( CabanaMD_ENABLE_Serial )
+  if( NOT Kokkos_ENABLE_SERIAL )
+    message( FATAL_ERROR "Requested SERIAL, but not enabled in Kokkos!" )
+  endif()
   add_definitions(-DCabanaMD_ENABLE_Serial=ON)
 endif()
 
-#option(CabanaMD_ENABLE_Pthread "Build CabanaMD with Pthread support" OFF)
-#if( CabanaMD_ENABLE_Pthread )
-#  add_definitions(-DCabanaMD_ENABLE_Pthread=ON)
-#  find_package(Threads)
-#endif()
+option(CabanaMD_ENABLE_Pthread "Build CabanaMD with Pthread support" OFF)
+if( CabanaMD_ENABLE_Pthread )
+  if( NOT Kokkos_ENABLE_PTHREAD )
+    message( FATAL_ERROR "Requested PTHREAD, but not enabled in Kokkos!" )
+  endif()
+  add_definitions(-DCabanaMD_ENABLE_Pthread=ON)
+endif()
 
 option(CabanaMD_ENABLE_OpenMP "Build CabanaMD with OpenMP support" OFF)
 if( CabanaMD_ENABLE_OpenMP )
-  add_definitions(-DCabanaMD_ENABLE_OpenMP=ON)
-  find_package(OpenMP)
-  if (OPENMP_FOUND)
-    set (CMAKE_C_FLAGS "${CMAKE_C_FLAGS} ${OpenMP_C_FLAGS}")
-    set (CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS} ${OpenMP_CXX_FLAGS}")
-    set (CMAKE_EXE_LINKER_FLAGS "${CMAKE_EXE_LINKER_FLAGS} ${OpenMP_EXE_LINKER_FLAGS}")
+  if( NOT Kokkos_ENABLE_OPENMP )
+    message( FATAL_ERROR "Requested OPENMP, but not enabled in Kokkos!" )
   endif()
+  add_definitions(-DCabanaMD_ENABLE_OpenMP=ON)
 endif()
 
 option(CabanaMD_ENABLE_Cuda "Build CabanaMD with Cuda support" OFF)
 if( CabanaMD_ENABLE_Cuda )
+  if( NOT Kokkos_ENABLE_CUDA )
+    message( FATAL_ERROR "Requested CUDA, but not enabled in Kokkos!" )
+  endif()
   add_definitions(-DCabanaMD_ENABLE_Cuda=ON)
```

Comment:
What does this `add_definitions` do?

Reply:
I'm keeping these to define one execution/memory space of those enabled in Kokkos. What do you think about switching to uppercase vs keeping the existing flags?

Reply:
Oh ok I get it now - those get passed to the compile line so you can access them within the implementation. You could just use the ones in Kokkos as `KOKKOS_ENABLE_CUDA` as well as the other device types will be accessible in your code. This is what I do in Cabana now. The other option is to use a `#cmakedefine` and a cmake config header file in which CMake will generate the macros to use in your application.

Reply:
I think I'll do this in another PR with a runtime option for backend. I had planned to leave these since I'm only ever using one backend at a time. #31 is adding the `cmakedefine`, but in the context of using CabanaMD as a library, it would make sense to match what Cabana is doing